src/Sarif.Viewer.VisualStudio/Controls/InfoBar.cs

```diff
@@ -197,5 +223,21 @@ public void OnActionItemClicked(IVsInfoBarUIElement infoBarUIElement, IVsInfoBar
                 this.clickAction?.Invoke(actionItem);
             }
         }
+
+        private static async Task AddInfoBarIfRequiredAsync(
+            ExceptionalConditions detectedConditions,
+            ExceptionalConditions individualCondition,
+            string message,
+            ImageMoniker imageMoniker)
+        {
+            // It's safe to access and update the dictionary because this method is called inside
+            // the write lock.
+            if ((detectedConditions & individualCondition) == individualCondition && !s_infoBarToConditionDictionary.Values.Contains(individualCondition))
+            {
+                var infoBar = new InfoBar(message, imageMoniker: imageMoniker);
+                await infoBar.ShowAsync();
+                s_infoBarToConditionDictionary.Add(infoBar, individualCondition);
```

Comment:
What's the implications of the ordering of these statements? Does it matter?

Reply:
It matters a tiny bit, and to the extent that it does, I have it wrong. If somehow the users closed the InfoBar that was just shown _before_ it was added to the list of tracked InfoBars (between lines 238 and 239), then no `InfoBar` for this `individualCondition` would every be shown again -- because every time we tried to show one, we'd see that it was already in the dictionary, and there'd be no way to remove it because it was already closed.

So I'll reverse 238 and 239.

---
In reply to: [527327149](https://github.com/microsoft/sarif-visualstudio-extension/pull/278#discussion_r527327149) [](ancestors = 527327149)